### PR TITLE
[DistillationTrainer refactor] Config cosmetic parity

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 39b (Group F) — stacked on 40a. Part of #6449. Cosmetic-only, no behaviour change.*

Drives `distillation_config.py` toward `grpo_config.py`'s exact structure (cosmetic diff-chasing is a core goal):
- Fields reordered into GRPO's 9 sections with `# Parameters that control …` phrasing: overridden-defaults, model+teacher, data-preprocessing, generation, vLLM (general / server / colocate — the three GRPO sub-groups), training, logging.
- `beta` → training, `disable_dropout` → model+teacher, `max_completion_length`/`ds3_gather_for_generation`/`shuffle_dataset`/`pad_to_multiple_of` → data-preprocessing — matching GRPO placement.
- `> [!NOTE]` overridden-defaults block (`learning_rate=1e-6`, `remove_unused_columns=False`); docstring Parameters section reordered to match the field order (`remove_unused_columns` now documented).

All 37 fields preserved (reorder asserts the field set is unchanged). Verified: ruff clean; config instantiates; docstring↔field consistency; suite 52 green.

---

> [!NOTE]
> **Low Risk**
> Documentation and field ordering only; defaults and validation are unchanged.
> 
> **Overview**
> Reorganizes **`DistillationConfig`** in `distillation_config.py` for structural parity with **`GRPOConfig`**, without changing defaults or runtime behavior.
> 
> The class docstring gains a **`> [!NOTE]`** block calling out overridden **`TrainingArguments`** defaults (`learning_rate=1e-6`, `remove_unused_columns=False`), and the **Parameters** section is reordered to follow the new field layout (including documenting **`remove_unused_columns`**).
> 
> Dataclass fields are grouped into nine GRPO-style sections with matching `# Parameters that control …` comments (overridden defaults, model+teacher, data preprocessing, generation, vLLM general/server/colocate, training, logging). Notable moves: **`beta`** → training, **`disable_dropout`** → model+teacher, and **`max_completion_length`** / **`ds3_gather_for_generation`** / **`shuffle_dataset`** / **`pad_to_multiple_of`** → data preprocessing. All 37 fields are preserved; only order and documentation wording shift slightly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3c1649b466299048835e33d16e7a8d8821416800. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_config.py [hunk 1]
```diff
@@ -41,34 +45,38 @@ class DistillationConfig(_BaseConfig):
...
             Keyword arguments passed to `AutoModelForCausalLM.from_pretrained` when instantiating the teacher model
             from a string.
+        disable_dropout (`bool`, *optional*, defaults to `False`):
+            Whether to disable dropout in the student model during training.
+
+        > Parameters that control the data preprocessing
+
+        remove_unused_columns (`bool`, *optional*, defaults to `False`):
+            Whether to only keep the columns used by the model's forward pass. Forced to `False` here: the trainer
+            consumes the raw prompt column and generates completions on-policy, so no columns are dropped.
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_config.py]: Fixed in c732c9b0, reworded the docstring and help to describe False as the default (it's a settable field; True is supported and keeps the prompt/image columns), matching GRPO's wording.

**Review by kashif — APPROVED** [review]
LGTM — cosmetic/config parity. Docstring/help fix for `remove_unused_columns` (default vs forced) addresses the earlier review note.
